### PR TITLE
Docs: Clarify package.json requirement in Getting Started (refs #13549)

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Clarifies that a `package.json` file is necessary to use `--init` in the Getting Started guide. (it errors currently when there is no `package.json` file present)

#### Is there anything you'd like reviewers to focus on?

Is this clear enough?

### CODE REVIEW

File: docs/user-guide/getting-started.md [hunk 1]
```diff
@@ -30,6 +30,8 @@ $ npx eslint --init
 $ yarn run eslint --init
 ```
 
+**Note:** `--init` assumes you have a `package.json` file already. If you don't, make sure to run `npm init` or `yarn create` beforehand.
```
Comment (mdjermanovic) [inline docs/user-guide/getting-started.md:33]: Should this be `yarn init` instead of `yarn create`?

Comment (nzakas) [reply docs/user-guide/getting-started.md:33]: Oops, yes

Comment (nzakas) [reply docs/user-guide/getting-started.md:33]:
```suggestion
**Note:** `--init` assumes you have a `package.json` file already. If you don't, make sure to run `npm init` or `yarn init` beforehand.
```

**Review by mdjermanovic — APPROVED** [review]
LGTM!

**Pull request merged**